buildSrc/src/main/groovy/org/gradle/modules/PatchExternalModules.groovy

```diff
@@ -20,48 +20,70 @@ import groovy.transform.CompileStatic
 import org.gradle.api.DefaultTask
 import org.gradle.api.artifacts.Configuration
 import org.gradle.api.file.CopySpec
+import org.gradle.api.file.FileCollection
 import org.gradle.api.internal.project.ProjectInternal
-import org.gradle.api.tasks.*
+import org.gradle.api.tasks.CacheableTask
+import org.gradle.api.tasks.Classpath
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.api.tasks.TaskAction
 
 @CacheableTask
 @CompileStatic
 class PatchExternalModules extends DefaultTask {
 
     @Internal
-    Configuration externalModules
+    Configuration modulesToPatch
 
     @Input
-    Set<String> getExternalModuleNames() {
-        externalModules.dependencies*.name as Set
+    Set<String> getNamesOfModulesToPatch() {
+        modulesToPatch.dependencies*.name as Set
     }
 
-    @Classpath
-    Configuration externalModulesRuntime
+    @Internal
+    Configuration allModules
+
+    @Input
+    Set<String> getFileNamesOfAllModules() {
+        allModules.incoming.artifacts*.file*.name as Set
+    }
+
+    @Internal
+    Configuration coreModules
+
+    @Input
+    Set<String> getFileNamesOfCoreModules() {
+        coreModules.incoming.artifacts*.file*.name as Set
```

Comment:
The latter doesn't exist. The manifest patcher code is looking at a `ResolvedDependency`, not a declared `Dependency`.